File: csharp/AoC2021/Day07/Day07/Program.cs
using Day07;

if (args.Length != 1)
{
    Console.Error.WriteLine("Usage: $0 <input file>");
    Environment.Exit(1);
}

var points = File
    .ReadAllLines(args[0])
    .SelectMany(line => line.Split(','))
    .Select(uint.Parse)
    .ToList();

var (optimalPoint1, fuelCost1) = Solution.FindOptimalPoint(points, Solution.ComputeFuelCostNaive)!.Value;
Console.WriteLine("Using the naive fuel computation, " +
                  $"the optimal point is {optimalPoint1}, " +
                  $"with a cost of {fuelCost1}.");

var (optimalPoint2, fuelCost2) = Solution.FindOptimalPoint(points, Solution.ComputeFuelCostCorrect)!.Value;
Console.WriteLine("Using the correct fuel computation, " +
                  $"the optimal point is {optimalPoint2}, " +
                  $"with a cost of {fuelCost2}.");

File: fsharp/AoC2021/Day07/Day07/Program.fs
open System
open System.IO

let args = Environment.GetCommandLineArgs()

if args.Length <> 2 then
    failwith "Usage: $0 <INPUT FILE>"

let input = File.ReadLines args.[1] |> Seq.head
let values = input.Split(',') |> Array.map uint

let point1, cost1 =
    values
    |> Solution.findOptimalPoint Solution.computeFuelNaive

printfn $"Using the naive fuel cost, the optimal point is {point1}, with a cost of {cost1}."

let point2, cost2 =
    values
    |> Solution.findOptimalPoint Solution.computeFuelCorrect

printfn $"Using the correct fuel cost, the optimal point is {point2}, with a cost of {cost2}."
